### PR TITLE
gc:orc defines gcArc

### PR DESCRIPTION
**Pull request by Clyybber**
https://github.com/nim-lang/Nim/pull/15811#discussion_r515871657
https://github.com/nim-lang/Nim/pull/17342#discussion_r592788519

### CODE REVIEW

File: tests/arc/torc_basic_test.nim [hunk 1]
```diff
@@ -136,3 +139,5 @@ main()
 main2()
 GC_fullCollect()
 echo "MEM ", getOccupiedMem()
+
+echo defined(gcArc)
```
Comment (timotheecour) [inline tests/arc/torc_basic_test.nim]: `doAssert defined(gcArc)`

https://nim-lang.github.io/Nim/contributing.html

> Use boolean conditions and doAssert for the testing by itself, don't rely on echo statements or similar;

(even though in the past we've relied on echo in tests)

Comment (Clyybber) [reply tests/arc/torc_basic_test.nim]: Done :)

**Comment by timotheecour** [comment]
* while you're at it, we should add a defined `gcRefc` for gc:refc
* this is a breaking change [1]; instead we should introduce `gcArcLike` instead, which wouldn't be a breaking change and is probably a bit easier to use (avoids things like `when defined(gcArc) and not defined(gcOrc)`, which also isn't future prof in case another arc variant is later introduced)
```nim
when defined(gcArc): echo 1
elif defined(gcOrc): echo 2
else: echo 3
```
before PR:
nim r --gc:orc main # 2
after PR:
nim r --gc:orc main # 1


(gcArcLike is better than gcArcOrc in case we later add another variant than orc based on arc, but we can bikeshed the name, eg gcSomeArc etc)

gcArcLike abides by the rule that each gc shall have its corresponding exclusive defined symbol

**Comment by Araq** [comment]
We have ``gcDestructors`` for ARC-like GCs already. :-)

**Pull request closed without merging**